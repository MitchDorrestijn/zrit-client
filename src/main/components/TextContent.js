/**
 * React related imports
 */
import React from 'react';
import {Col} from 'reactstrap';

/**
 * Style related imports
 */
import '../css/textblock.css';

/**
 * This component renders the textblock above the CRUD pages
 */
const TextContent = (props) => {
  return (
		<Col className="textblock" sm="12" md={{
      size: 6,
      offset: 3
    }}>
	    <h2>{props.title}</h2>
	    <p>{props.children}</p>
    	<hr/>
  	</Col>
	);
}

export default TextContent;
