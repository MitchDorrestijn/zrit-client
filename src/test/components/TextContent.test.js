//The tests for the TextContent component are defined here (TextContent can be found on the TablePage)

import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import TextContent from '../../main/components/TextContent';

const wrapper = shallow(<TextContent />);

test('<TextContent /> should match snapshot', () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('The class .textblock should be applied in a <Col />', () => {
  expect(wrapper.find('.textblock').text()).toBe('<Col />')
});

test('<TextContent /> render one <Col />', () => {
  expect(wrapper.find('Col').length).toBe(1);
});

test('<TextContent /> render one <h2></h2>', () => {
  expect(wrapper.find('h2').length).toBe(1);
});

test('<TextContent /> render one <p></p>', () => {
  expect(wrapper.find('p').length).toBe(1);
});

test('<TextContent /> render one <hr />', () => {
  expect(wrapper.find('hr').length).toBe(1);
});
