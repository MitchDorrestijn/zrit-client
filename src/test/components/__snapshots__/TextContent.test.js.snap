// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextContent /> should match snapshot 1`] = `
<Col
  className="textblock"
  md={
    Object {
      "offset": 3,
      "size": 6,
    }
  }
  sm="12"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <h2 />
  <p />
  <hr />
</Col>
`;
